pkg/model/project.go

```diff
@@ -442,3 +472,56 @@ func (p *Project) GetAllRBACRoles() []*ProjectRBACRole {
 
 	return all
 }
+
+// isBuiltinRBACRole checks whether the name is the name of built-in role.
+func isBuiltinRBACRole(name string) bool {
+	return name == builtinRBACRoleAdmin.String() ||
+		name == builtinRBACRoleEditor.String() ||
+		name == builtinRBACRoleViewer.String()
+}
+
+// AddRBACRole adds a custom RBAC role.
+func (p *Project) AddRBACRole(name string, policies []*ProjectRBACPolicy) error {
+	if p.HasRBACRole(name) {
+		return fmt.Errorf("the name of %s is already used", name)
+	}
+	if isBuiltinRBACRole(name) {
+		return fmt.Errorf("the name of built-in role cannot be used")
+	}
+	p.RbacRoles = append(p.RbacRoles, &ProjectRBACRole{
+		Name:     name,
+		Policies: policies,
+	})
+	return nil
+}
+
+// UpdateRBACRole updates a custom RBAC role.
+// Built-in role cannot be updated.
+func (p *Project) UpdateRBACRole(name string, policies []*ProjectRBACPolicy) error {
+	for _, v := range p.RbacRoles {
+		if v.Name == name {
+			v.Policies = policies
+			return nil
+		}
+	}
+	if isBuiltinRBACRole(name) {
+		return fmt.Errorf("built-in role cannot be updated")
+	}
+	return fmt.Errorf("%s role doesn't exist", name)
+}
+
+// DeleteRBACRole deletes a custom RBAC role.
+// Built-in role cannot be deleted.
+func (p *Project) DeleteRBACRole(name string) error {
+	for i, v := range p.RbacRoles {
+		if v.Name == name {
+			c := copy(p.RbacRoles[i:], p.RbacRoles[i+1:])
+			p.RbacRoles = p.RbacRoles[:i+c]
+			return nil
+		}
+	}
+	if isBuiltinRBACRole(name) {
+		return fmt.Errorf("built-in role cannot be deleted")
+	}
+	return fmt.Errorf("%s role doesn't exist", name)
```

Comment:
nit: just want to write the message without using any special character in the code.

```suggestion
	return fmt.Errorf("%s role does not exist", name)
```